Khan/wonder-blocks
# WB-1652.3: Add theming support to ModalHeader and ModalFooter

## Summary:

Following up on #2141, this PR focuses on adding theming support to the
remaining modal building blocks: ModalHeader and ModalFooter.

The main change is to add the "eggplant" bg color to ModalHeader and to refactor
these components to use media queries instead of `MediaLayout`.

Also added better docs for ModalPanel and ModalFooter.

Issue: WB-1652

## Test plan:

1. Verify that the `ModalHeader` component is now using the `eggplant` bg color.

2. Verify that the `ModalHeader` and `ModalFooter` docs look correct.
